### PR TITLE
add channel and messages enumerators

### PR DESCRIPTION
**Pull request by ryanfkeepers**
adds enumerators for channels, and a delta
of channel messages.
Also updates the its struct to include group ids, and reformats some of the struct schema.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :sunflower: Feature

#### Issue(s)

* #3989

#### Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/pkg/services/m365/api/channels_pager.go [hunk 1]
```diff
@@ -1,43 +1,223 @@
...
+			continue
+		}
+
+		if err != nil {
+			return nil, DeltaUpdate{}, graph.Wrap(ctx, err, "retrieving page of channels")
+		}
+
+		vals, err := pager.ValuesIn(page)
+		if err != nil {
+			return nil, DeltaUpdate{}, graph.Wrap(ctx, err, "extracting channels from response")
```
Comment (ashmrtn) [inline src/pkg/services/m365/api/channels_pager.go:109]: nit: should this be `extracting channel messages`?

File: src/pkg/services/m365/api/channels_pager.go [hunk 2]
```diff
@@ -1,43 +1,223 @@
...
+		}
+
+		vs = append(vs, vals...)
```
Comment (ashmrtn) [inline src/pkg/services/m365/api/channels_pager.go:112]: do we need to handle deleted messages in delta results?

File: src/pkg/services/m365/api/channels_pager.go [hunk 3]
```diff
@@ -1,43 +1,223 @@
...
+		vs               = []models.ChatMessageable{}
+		pager            = c.NewChannelMessageDeltaPager(teamID, channelID)
+		invalidPrevDelta = len(prevDelta) == 0
+		newDeltaLink     string
+	)
+
+	// Loop through all pages returned by Graph API.
+	for {
+		page, err := pager.GetPage(graph.ConsumeNTokens(ctx, graph.SingleGetOrDeltaLC))
+		if graph.IsErrInvalidDelta(err) {
```
Comment (ashmrtn) [inline src/pkg/services/m365/api/channels_pager.go:93]: hmmm I know this probably follows similar code we have, but looking at it there's the assumption that we get the invalid delta token error on the first call to the API. It's probably fine to just check for `IsErrInvalidDelta`, but we should probably clear `vs` when we see it

File: src/pkg/services/m365/api/channels_pager.go [hunk 4]
```diff
@@ -1,43 +1,223 @@
...
+
+func (p *channelPageCtrl) GetPage(
+	ctx context.Context,
+) (PageLinker, error) {
+	resp, err := p.builder.Get(ctx, p.options)
+	return resp, graph.Stack(ctx, err).OrNil()
+}
+
+func (p *channelPageCtrl) GetOdataNextLink() *string {
+	return ptr.To("")
```
Comment (ashmrtn) [inline src/pkg/services/m365/api/channels_pager.go:161]: is this intentionally returning "no next page"? If it is could you please add a comment that states why it's alright to assume there's no next page

**Review by ashmrtn — APPROVED** [review]
mostly nits and maybe one or two things that were missed

File: src/pkg/services/m365/api/channels_pager.go (hunk 2 above)
Comment (ryanfkeepers) [reply src/pkg/services/m365/api/channels_pager.go:112]: We aren't persisting metadata yet, so it's all non-incremental through the incremental pager.  Proper incremental handling will arrive in v1.

Also, I'm planning to do a refactor between v0 and v1 that centralizes the delta page runners (we're halfway there already) which should resolve this concern on its own.

File: src/pkg/services/m365/api/channels_pager.go (hunk 4 above)
Comment (ryanfkeepers) [reply src/pkg/services/m365/api/channels_pager.go:161]: Hah, no, not intended.  This is copied code which slipped through my attention.  Good catch, ty!

File: src/pkg/services/m365/api/channels_pager.go (hunk 3 above)
Comment (ryanfkeepers) [reply src/pkg/services/m365/api/channels_pager.go:93]: Ah, good call, thank you.

**Pull request merged**